apps/src/studio/StudioAnimation.js

```diff
@@ -0,0 +1,202 @@
+  // the current frame changing at the framerate.
+  this.setAnimationRate(utils.valueOr(options.animationRate,
+      constants.DEFAULT_ANIMATION_RATE));
+};
+
+/**
+ * Test only function so that we can start our id count over.
+ */
+StudioAnimation.__resetIds = function () {
+  uniqueId = 0;
+};
+
+/** @returns {SVGImageElement} */
+StudioAnimation.prototype.getElement = function () {
+  return this.element_;
+};
+
+/**
+ * Create an image element with a clip path
+ */
+StudioAnimation.prototype.createElement = function (parentElement) {
+  var nextId = (uniqueId++);
+
+  // create our clipping path/rect
+  this.clipPath_ = document.createElementNS(SVG_NS, 'clipPath');
+  var clipId = 'studioanimation_clippath_' + nextId;
+  this.clipPath_.setAttribute('id', clipId);
+  var rect = document.createElementNS(SVG_NS, 'rect');
+  rect.setAttribute('width', this.spriteSheet_.frameWidth * this.renderScale_);
+  rect.setAttribute('height', this.spriteSheet_.frameHeight * this.renderScale_);
+  this.clipPath_.appendChild(rect);
+  parentElement.appendChild(this.clipPath_);
+
+  var itemId = 'studioanimation_' + nextId;
+  this.element_ = document.createElementNS(SVG_NS, 'image');
+  this.element_.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:href',
+     this.spriteSheet_.assetPath);
+  this.element_.setAttribute('id', itemId);
+  this.element_.setAttribute('height',
+      this.spriteSheet_.assetHeight() * this.renderScale_);
+  this.element_.setAttribute('width',
+      this.spriteSheet_.assetWidth() * this.renderScale_);
+  parentElement.appendChild(this.element_);
+
+  this.element_.setAttribute('clip-path', 'url(#' + clipId + ')');
+};
+
+/**
+ * Remove our element/clipPath/animator
+ */
+StudioAnimation.prototype.removeElement = function() {
+
+  if (this.element_) {
+    this.element_.parentNode.removeChild(this.element_);
+    this.element_ = null;
+  }
+
+  // remove clip path element
+  if (this.clipPath_) {
+    this.clipPath_.parentNode.removeChild(this.clipPath_);
+    this.clipPath_ = null;
+  }
+
+  if (this.animator_) {
+    window.clearInterval(this.animator_);
+    this.animator_ = null;
+  }
+};
+
+/**
+ * Display the current frame at the given location
+ */
+StudioAnimation.prototype.redrawCenteredAt = function (center) {
+  var frame = this.spriteSheet_.getFrame(this.currentAnimation_, this.currentFrame_);
+  var scale = this.renderScale_;
+
+  // Preserved behavior: When scaling a sprite up, we actually scale around the
+  //       bottom-center of the sprite (so feet stay planted in the same place)
+  //       rather than actually around its center.
+  //       That's what the (2 * scale - 1) bit is about; just change that to
+  //       (scale) if you want to scale about the sprite center again.
+  // TODO: Improve this by scaling around an explicitly encoded 'sprite center'
+  var topLeft = {
+    x: center.x - (frame.width / 2) * scale,
+    y: center.y - (frame.height / 2) * (2 * scale - 1)
+  };
```

Comment:
Found a more elegant way to preserve this behavior.

What we're doing here is scaling X around the sprite center, while scaling Y around what would be the bottom edge of the sprite at its original size.  This is useful for infinity and hoc2015 because the characters' feet stays in about the right position as they scale.

For proof that this formula works, [see this graph](https://www.desmos.com/calculator/hdxo1uff36).  `x` represents the current scale, while `l` `r` `t` `b` are mapped to the y-axis and represent the final left, right, top, and bottom of the sprite, respectively. `i` and `j` sliders stand in for center.x and center.y. The `c` slider is both frame.width and frame.height (we always seem to use square sprites).

![screenshot from 2015-10-23 03 11 36](https://cloud.githubusercontent.com/assets/1615761/10690006/ddf50d68-7933-11e5-9c25-24c364f45c7d.png)

This trivial case is a 1 unit square sprite centered at (0, 0).  Notice that at `scale`=1 the sprite is at its original size and centered around 0 in both dimensions (hence the two intersection points).  Now, the red and green lines (left and right) are mirrored across y=0 because the sprite should always be horizontally centered.  The purple and orange lines behave differently - their difference changes at the same rate as the horizontal case, but the bottom of the sprite, the orange line, never moves.